scripts/src/build.sh

```diff
@@ -45,7 +47,47 @@ examples_build() {
   TARGET_DIR="$2"
   shift
   shift
-  cd "$ROOT_DIR"/examples
-  CARGO_TARGET_DIR="$TARGET_DIR" cargo +nightly hack build --release --workspace "$@"
-  cd "$ROOT_DIR"
+
+
+  if [ -n "$1" ]
+  then
+    has_yamls=$(echo "$1" | grep "yamls=" || true)
+  else
+    has_yamls=""
+  fi
+
+  if  [ -n "$has_yamls" ]
+  then
+    if ! command -v perl &> /dev/null
```

Comment:
Although one new dependency is added the main concern is that yamls parsed here with perl and in gtest. So there will be two places with parsing logic: this may lead to maintain costs in the future.

Can't suggest anything useful here: one crazy idea is to add parameter to gtest so it returns the list of all involved examples. Another one is to extract common code for parsing and create additional binary that parses yamls and returns the required list.

Reply:
Oh, I suppose we will spend much more time to do that, then to support two small codes for yaml parsing.
Also, as you now gtest use only compiled wasm files, but build.sh need to now sources locations.
So, my position is: this functionality is only for some developers needs and we can skip architecture design here.